src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -1603,6 +1603,88 @@ Copyright (C) Microsoft Corporation. All rights reserved.
+    Calls the GetCompatiblePlatform task to determine the most compatible platform for each project.
+    This prevents overbuilding a project when 'AnyCPU' is available, for example.
+
+    ======================================================================================
+  -->
+
+  <!-- Managed projects need to have PlatformTarget set for SetPlatform negotiation. Default to $(Platform), which is AnyCPU by default. -->
+  <PropertyGroup>
+    <PlatformTarget Condition="'$(EnableDynamicPlatformResolution)' == 'true' and '$(PlatformTarget)' == ''
+                                and '$(MSBuildProjectExtension)' != '.vcxproj' and '$(MSBuildProjectExtension)' != '.nativeproj'">$(Platform)</PlatformTarget>
+  </PropertyGroup>
+
+  <Target Name="_GetProjectReferencePlatformProperties"
+          Condition="'$(EnableDynamicPlatformResolution)' == 'true'
+                     and '$(BuildingInsideVisualStudio)' != 'true'
+                     and '@(_MSBuildProjectReferenceExistent->Count())' != '0'">
+
+    <!-- Allow preset SetPlatform to override this operation -->
+    <ItemGroup>
+      <_MSBuildProjectReferenceExistent Condition="'%(_MSBuildProjectReferenceExistent.SetPlatform)' != ''">
+        <SkipGetPlatformProperties>true</SkipGetPlatformProperties>
+      </_MSBuildProjectReferenceExistent>
+    </ItemGroup>
+
+    <ItemGroup>
+      <_ProjectReferencePlatformPossibilities Include="@(_MSBuildProjectReferenceExistent)" 
+                                              Condition="'%(_MSBuildProjectReferenceExistent.SkipGetPlatformProperties)' != 'true'"/>
+    </ItemGroup>
+
+    <!-- Managed Platform "source of truth" is $(PlatformTarget). For cpp it's $(Platform) -->
+    <PropertyGroup>
+      <ParentPlatform>$(PlatformTarget)</ParentPlatform>
+      <ParentPlatform Condition="'$(MSBuildProjectExtension)' == '.vcxproj' or '$(MSBuildProjectExtension)' == '.nativeproj'">$(Platform)</ParentPlatform>
+    </PropertyGroup>
+
+    <!-- If we're looking at a c++ project from a managed project, map managed platforms to native platforms. -->
+    <ItemGroup>
+      <_ProjectReferencePlatformPossibilities Condition="'$(MSBuildProjectExtension)' != '.vcxproj' and '$(MSBuildProjectExtension)' != '.nativeproj' and '%(_ProjectReferencePlatformPossibilities.IsVcxOrNativeProj)' == 'true'">
+        <!-- Taken from https://docs.microsoft.com/visualstudio/msbuild/assignprojectconfiguration-task -->
+        <PlatformLookupTable Condition="$(PlatformLookupTable) == ''">AnyCPU=Win32;x86=Win32</PlatformLookupTable>
+      </_ProjectReferencePlatformPossibilities>
+    </ItemGroup>
+
+    <Warning Condition="'%(_ProjectReferencePlatformPossibilities.PlatformOptions)' == ''"
+           Code="MSB3982" 
+           Text="EnableDynamicPlatformResolution is enabled, but '%(_ProjectReferencePlatformPossibilities.Identity)' has no 'Platforms' set. It will be built without a specified platform."/>
```

Comment:
Can you move this into the task so it can be localized?

Reply:
Opted for this to avoid an unnecessary task call. We have other `Warning` calls that have codes and non-translated text. Out of curiosity, what's the deciding factor there? #legacy?

Will pop this into the task regardless.

Reply:
All of our user-facing errors should be localized. We can sometimes get away with an unlocalized string if it's "something's horked with your installation" or "this condition should never happen" type stuff, plus some stuff has just Been Like That.